query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala

```diff
@@ -65,6 +65,11 @@ final case class PartKeysDistConcatExec(queryContext: QueryContext,
 final case class LabelValuesDistConcatExec(queryContext: QueryContext,
                                            dispatcher: PlanDispatcher,
                                            children: Seq[ExecPlan]) extends MetadataDistConcatExec
+final case class LabelNamesDistConcatExec(queryContext: QueryContext,
+                                           dispatcher: PlanDispatcher,
+                                           children: Seq[ExecPlan]) extends DistConcatExec {
+  addRangeVectorTransformer(LimitFunctionMapper(1))
```

Comment:
Let us not add transformers during runtime. Can implement this in do/execute methods.